apibara/dna
# charts: add operator helm chart

### Summary

The Apibara Operator is the easiest way to run Apibara indexers on Kubernetes.
The operator doesn't have any complex dependencies or requirements, but
requires configuring the service account correctly which can be difficult to
do.

### Testing strategy

- Start a `minikube` cluster.
- Create a namespace for the tests, e.g. `kubectl create namespace test-ns`.
- Install the chart, enabling tests. Notice you also need an Apibara API Key if
  you want the indexer to run properly.

```
helm install my-test oci://quay.io/apibara-charts/operator \
    --namespace=test-ns \
    --set-json 'customArgs=["start", "--namespace=test-ns"]' \
    --set-json='test.enabled=true' \
    --set-string='test.apiKey=dna_xxx'
```

- Verify that the indexer is running correctly. `kubectl get pods -n test-ns`.
